.github/workflows/go.yml

```diff
@@ -43,3 +43,47 @@ jobs:
     - name: Build
       working-directory: ./scripts
       run: make golangci
+
+  check-changes:
+    name: Check whether tests need to be run based on diff
+    runs-on: [ubuntu-latest]
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - uses: antrea-io/has-changes@v2
+      id: check_diff
+      with:
+        paths: scripts/*
+    outputs:
+      has_changes: ${{ steps.check_diff.outputs.has_changes }}
+
+  test-unit:
+    needs: check-changes
+    if: ${{ needs.check-changes.outputs.has_changes == 'yes' }}
+    name: Unit test
+    runs-on: [ubuntu-latest]
+    steps:
+    - name: Check-out code
+      uses: actions/checkout@v2
+    - name: Set up Go using version from go.mod
+      uses: actions/setup-go@v2
```

Comment:
update to the latest version of these actions: `actions/checkout@v3` and `actions/setup-go@v4`.
In particular `actions/setup-go@v4` supports caching out of the box:
```yaml
    - name: Set up Go using version from go.mod
      uses: actions/setup-go@v4
      with:
        go-version-file: 'go.mod'
        cache-dependency-path: 'scripts/go.sum'
```
That means you can get rid of the custom caching logic

Reply:
Please update these 2 actions for all existing workflows, not just the new one